executor/point_get.go

```diff
@@ -332,7 +338,37 @@ func (e *PointGetExecutor) get(ctx context.Context, key kv.Key) ([]byte, error)
 		}
 		// fallthrough to snapshot get.
 	}
-	return e.snapshot.Get(ctx, key)
+
+	cacheDB := kv.Cache()
+
+	tableID, _, _, err := tablecodec.DecodeKeyHead(key)
+	if err != nil {
+		return e.snapshot.Get(ctx, key)
+	}
+
+	isLock, err := cacheDB.IsLock(tableID)
+	if err != nil {
+		return e.snapshot.Get(ctx, key)
+	}
+
+	if !isLock {
+		return e.snapshot.Get(ctx, key)
+	}
+
+	val, err = cacheDB.Get(ctx, tableID, key)
+	// key does not exist then get from snapshot and set to cache
+	if err != nil {
```

Comment:
Should we check the error type? Or it will only be one error.

Reply:
There are two errors will be return. I will fix it